packages/react/runtime/__test__/lifecycle.test.jsx

```diff
@@ -200,7 +199,7 @@ describe('componentWillUnmount', () => {
       }
 
       componentDidMount() {
```

Comment:
We should also add a test case of `componentDidUpdate` here to ensure it is "early" enough to get the correct state.

Before:

1st setState -> render ->  wait a long time (other 2nd setState may happen) -> componentDidUpdate get 2nd state

After: 

setState -> render -> componentDidUpdate  get the 1st state

Reply:
I've added tests for `componentDidUpdate`:

- 0543e71 adds the tests that can pass on `main` branch
- 7d9ddf2 updates the tests (which shows `componentDidUpdate` can get correct state when there are multiple updates before main-thread callback)